### ISSUE TITLE
Support for control lora

### ISSUE THREAD

**Issue by adhikjoshi** [huggingface/diffusers#4679]
There's new control lora release, 

https://huggingface.co/stabilityai/control-lora?s=09

How can we use it with diffusers?

**Comment by sayakpaul** [comment]
You cannot yet.

**Comment by haofanwang** [comment]
Any timeline for supporting it?

**Comment by MaxTran96** [comment]
any timeline for supporting this?

**Comment by yanchaoguo** [comment]
+1

**Comment by patrickvonplaten** [comment]
Working on it!

**Comment by patrickvonplaten** [comment]
@sayakpaul do you have bandwitdh for it? Otherwise I can look into picking it up :-)

**Comment by HighCWu** [comment]
I implemented a `control-lora` training code based on diffusers' controlnet. The idea is basically the same as stabilityai's. I found `diffusers` clear the lora layer of attention layers when xformers is enabled, so I did a little hack. At present, I have only tested the training and inference on sd v1.5. In theory, it can be compatible with sdxl's control lora by making some modifies. If anyone is interested, you could refer to my code to modify a sdxl version.
In addition, because I proposed my own concept of control-lora at the beginning of the year, so this new implementation becomes version 2, which is compatible with `ControlNetModel` in diffusers but need a `tie_weights` step.
Here is my repo [HighCWu/control-lora-v2](https://github.com/HighCWu/control-lora-v2).

**Comment by patrickvonplaten** [comment]
It's being worked on: https://github.com/huggingface/diffusers/pull/4899

**Comment by HighCWu** [comment]
> It's being worked on: #4899

Great work

**Comment by yanchaoguo** [comment]
> It's being worked on: #4899
Please make it merge into the main branch as soon as possible

**Comment by sayakpaul** [comment]
@HighCWu thanks for bringing your implementation to our attention. From what I can see from [here](https://github.com/HighCWu/control-lora-v2/blob/master/models/controllora.py), we're doing exactly the same in https://github.com/huggingface/diffusers/pull/4899. If you have any comments there let me know :)

**Comment by yanchaoguo** [comment]
> It's being worked on: #4899
No effect! @sayakpaul @patrickvonplaten @HighCWu 
please，has this problem been solved:
【Results don't seem to be following the conditioning image at all. I will play with the hyperparameters a bit to see how the results get affected】

**Comment by MaxTran96** [comment]
Any eta on when this will be done?

**Comment by patrickvonplaten** [comment]
@sayakpaul do you want to give this another try or open it for the community?

**Comment by sayakpaul** [comment]
@patrickvonplaten https://github.com/huggingface/diffusers/issues/5800

**Comment by alexblattner** [comment]
Bump

**Comment by alexblattner** [comment]
bump

**Comment by Astropulse** [comment]
As there's been activity recently, is this any closer to being added?

Regular controlnets are massive and heavy, support for controlloras is kind of a big deal.

**Comment by sayakpaul** [comment]
Well, this is still a LoRA on top of a ControlNet (initialized from the UNet). So, it's still the ControlNet pipeline. So, I don't think the computational cost here is amortized. I have explained the difference in details:
https://github.com/huggingface/diffusers/pull/5938#issuecomment-2492680106

We're adding support for Control LoRA that was released by Black Forest Labs recently:
https://github.com/huggingface/diffusers/pull/9999

And a training script to top it off:
https://github.com/huggingface/diffusers/pull/10130

**Comment by Astropulse** [comment]
But will you be adding support for SAI's control loras for stable diffusion 1.5?

**Comment by sayakpaul** [comment]
1. Yes, we will which is why @a-r-r-o-w has assigned the issue. 
2. My point was about Control Net LoRA from SAI not giving any memory/computational costs improvement. 
3. Those LoRA are for SDXL and not for SD v1.5.

**Comment by Astropulse** [comment]
The comfyui extraction + inference methods for SAI's control loras are not exclusive to sdxl, they also work for creating 1.5 control loras from controlnet checkpoints: [code](https://github.com/Stability-AI/stability-ComfyUI-nodes/blob/master/control_lora_create.py). There is a significant memory and storage advantage to using control loras instead of full controlnet checkpoints, or the "subtracted" controlnet models.

**Comment by sayakpaul** [comment]
Ah that is a new for me, I wasn't aware of this one. Thanks for flagging. Possible to open a new issue for this extracted type of LoRAs so that we could better track? Maybe if you have a checkpoint that you know to be working well, that would be great to have as well.

**Comment by Astropulse** [comment]
Sure if thats necessary, I believe they are the same as the control loras mentioned above (the control_lora_create script was uploaded at nearly the same time as the sdxl control loras).

Let me know if I should open a new issue.

**Comment by lavinal712** [comment]
![Image](https://github.com/user-attachments/assets/95706364-2bff-49d2-ab21-9c39f926a57e)

**Comment by sayakpaul** [comment]
Brilliant work!
